### PR TITLE
syncserver: remove pkgs/module/dependencies

### PR DESCRIPTION
**Pull request by Mic92**
Syncserver is python2-only and there are no plans to upgrade to python3
as upstream works on a new Rust implementation.
It broke in nixpkgs because some dependencies became python3-only.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by jonringer — APPROVED** [review]
LGTM

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

File: nixos/modules/rename.nix [hunk 1]
```diff
@@ -82,6 +82,8 @@ with lib;
     # rmilter/rspamd
     (mkRemovedOptionModule [ "services" "rmilter" ] "Use services.rspamd.* instead to set up milter service")
 
+    (mkRemovedOptionModule [ "services" "firefox" "syncserver" ] "Syncserver uses out-dated python2 libraries. Upstream works on new Rust implementation")
```
Comment (Mic92) [inline nixos/modules/rename.nix]: I think, this one is still wrong.

**Comment by FRidh** [comment]
cc @Nadrieril

**Comment by Nadrieril** [comment]
Ah well, that sad :( ut if it's broken and upstream is not working on Python3, then I guess we have no choice

**Comment by Infinisil** [comment]
Well there is the possibility of introducing older versions of those packages, so we're not choiceless, but not sure how good that choice would be.

**Comment by FRidh** [comment]
This package was up for removal several times. @Nadrieril then took up maintainership and kept it working. If you're also willing to maintain the Python 2 packages it depends upon, I suppose it could stay (though Python2 *may* eventually get marked as insecure).

**Comment by Nadrieril** [comment]
No, as much as I would like to, I don't have the bandwith to maintain a couple of outdated python libraries and all of their dependencies

**Comment by andir** [comment]
Since it is probably relevant: What are you migrating to if you are no longer using the synserver module? I guess moving to the mozilla provided service is not an option as otherwise we would never have considered self-hosting?

**Comment by Mic92** [comment]
Mozilla's new implementation could not be build yet a few months ago. There was one rust library not yet opensourced by google. This could be different now. Apart from that this would be the way to go in future.

**Comment by Nadrieril** [comment]
I tried to make it build by rolling back a bunch of python module versions, but I hit https://github.com/NixOS/nixpkgs/issues/71824 that I have no clue how to fix. I'm ok with removing syncserver altogether, and waiting for the new rust implementation to be usable :(

**Comment by dasj19** [comment]
@Nadrieril could we get the progress you made? I have also tried something and hit the webtest issue. But I'm willing to give a second shot at it.

**Comment by Nadrieril** [comment]
What I did is I tried to build syncserver on the latest 20.03, and whenever there was a dependency issue I rolled back that dependency to a commit where syncserver used to build. After 4-5 deps I hit #71824, but not just in webtest. A few other packages had issues importing paste.deploy.
Sadly I did not keep this around but it shouldn't be too hard to get to the same state yourself

**Comment by Mic92** [comment]
Too lazy to rebase this another time...

**Pull request closed without merging**